python: add note to zero-code docs highlighting the need for distro package (#4912)
open-telemetry/opentelemetry.io: content/en/docs/zero-code/python/_index.md

The `opentelemetry-distro` package installs the API, SDK, and the
`opentelemetry-bootstrap` and `opentelemetry-instrument` tools.

{{% alert title="Note" color="info" %}}

You must install a distro package to get auto instrumentation working. The
`opentelemetry-distro` package contains the default distro to automatically
configure some of the common options for users. For more information, see
[OpenTelemetry distro](/docs/languages/python/distro/).

{{% /alert %}}

> 

Author: So, is your suggestion not to add the note? 
Although we have in all examples the command to install the opentelemetry-distro package, it may still not be so clear to all users that they need a distro package to get auto-instrumentation working. 

Example here: https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2157

> 

Reviewer: Apologies, I missed that open discussion before merging the PR

> 

Reviewer: Let's leave it for now, but I think we need to do a major review of the python docs here, the "distro" page talks a lot about automatic instrumentation, so it should probably be moved over into the Zero-code section and be reworked

> 

Author: @svrnm will create an issue to track this

The `opentelemetry-bootstrap -a install` command reads through the list of
packages installed in your active `site-packages` folder, and installs the
corresponding instrumentation libraries for these packages, if applicable. For
example, if you already installed the `flask` package, running
`opentelemetry-bootstrap -a install` will install
`opentelemetry-instrumentation-flask` for you.

> {{% alert title="Note" color="info" %}}
> 
> If you leave out `-a install`, the command will simply list out the recommended
> instrumentation libraries to be installed. More information can be found
> [here](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/opentelemetry-instrumentation#opentelemetry-bootstrap).
> 
> {{% /alert %}}

Reviewer 2: I'd propose that this be in plain text, not a note, given that the previous paragraph describes the command as well. How about:

```suggestion
Running `opentelemetry-bootstrap` without arguments lists the recommended
instrumentation libraries to be installed. For more information, see
[`opentelemetry-bootstrap`](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/opentelemetry-instrumentation#opentelemetry-bootstrap).
```

The `opentelemetry-distro` package installs the API, SDK, and the
`opentelemetry-bootstrap` and `opentelemetry-instrument` tools.

> **NOTE:** You need to install a distro package to get auto instrumentation

> > working. The `opentelemetry-distro` package contains the default distro and
> > configurator to automatically configures some of the common options for users.

Reviewer 3: ```suggestion
> working. The `opentelemetry-distro` package contains the default distro to
> automatically configure some of the common options for users.
```

> > **NOTE:** You need to install a distro package to get auto instrumentation
> > working. The `opentelemetry-distro` package contains the default distro and
> > configurator to automatically configures some of the common options for users.
> > For more information about `opentelemetry-distro` check
> > [here](/docs/languages/python/distro/).

Reviewer 4: Could you also convert this Markdown note to short code? Here's an [example](https://github.com/open-telemetry/opentelemetry.io/blob/ada8fd7a9d47e188f5e7c94e7d9affd7c0588c5e/content/en/docs/zero-code/js/_index.md?plain=1#L48). And while we're at it, we should change the second note on the page as well.

```suggestion
> **NOTE:** You need to install a distro package to get auto instrumentation
> working. The `opentelemetry-distro` package contains the default distro and
> configurator to automatically configures some of the common options for users.
> For more information about `opentelemetry-distro` check
> [here](/docs/languages/python/distro/).
```

Author: Thanks for the review. Will update the notes then
